frontend/src/components/AppPercentage.vue

```diff
@@ -71,4 +91,35 @@ const d = computed(() => {
     stroke-width: 10px;
   }
 }
+
+.bar {
+  position: relative;
+  flex-grow: 0;
+  flex-shrink: 0;
+  width: 90px;
+  height: 30px;
+  font-size: 0.9rem;
+
+  & > * {
+    display: flex;
+    position: absolute;
+    top: 0;
+    left: 0;
+    align-items: center;
+    justify-content: center;
+    width: 100%;
+    height: 100%;
+  }
+
+  rect {
+    fill: none;
+    stroke: $light-gray;
+    stroke-width: 3px;
+    &.fill {
+      // fill: $theme;
+      fill: $theme-mid;
```

Comment:
There is probably not enough contrast between `$theme-mid` and the white background or gray outline. Instead, you should probably use the darker `$theme`, as the ring mode does, and just put the text % above or to the side. It's not the most ideal for saving space, but it would be the most accessible.

If you wanted to keep the text on top of the bar and use `$theme`, you'd have to make use of `mix-blend-mode` to make sure any portion of the text over `$theme` is white and any portion over white/light gray is black.

Reply:
I did try and play with this for a while, but unfortunately most blending modes available didn't really work out (at best, it could be black over grey, and orange over the teal...)

Will probably just leave it as text off to the left.